### PR TITLE
[Javadocs] add to o.o.action.admin

### PR DESCRIPTION
**Pull request by nknize**
Adds javadocs to classes in package o.o.action.admin and removes unused classes. Also 
adds custom javadoc tags `opensearch.internal`, `opensearch.experimental`, and `opensearch.api`

### CODE REVIEW

File: server/src/main/java/org/opensearch/action/admin/indices/upgrade/post/ShardUpgradeRequest.java [hunk 1]
```diff
@@ -1,65 +0,0 @@
-/*
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/admin/indices/upgrade/post/ShardUpgradeRequest.java:1]: @nknize do you think we could move these changes to separate pull request (as you did with aliases & types)?

Comment (nknize) [reply server/src/main/java/org/opensearch/action/admin/indices/upgrade/post/ShardUpgradeRequest.java:1]: Sure thing!

**Review by reta — APPROVED** [review]
